### PR TITLE
Document Contains OPENJSON breaking change

### LINKED ISSUE #4423: Document break to Contains if the SQL Server compatibility level is too low
Users running on really old (unsupported) versions of SQL Server - as well as those who have set the compatibility level to a low value - will break when upgrading to EF Core 8.0, since the translation of Contains now relies on OPENJSON (see https://github.com/dotnet/efcore/issues/31323 for an example). We should add a breaking change note for this.

The same happens if using SQLite versions older than 3.38.0. This is more rare as usually people use the bundled SQLite, but it could happen.

### PR DESCRIPTION
**Pull request by roji**
Closes #4423

### CODE REVIEW

File: entity-framework/core/what-is-new/ef-core-8.0/breaking-changes.md [hunk 1]
```diff
@@ -14,8 +14,69 @@ This page will document API and behavior changes that have the potential to brea
...
+
+#### Mitigations
+
+If your database is SQL Server 2016 (13.x) or newer, or if you're using Azure SQL, check the configured compatibility level of your database via the following command:
+
+```sql
+SELECT name, compatibility_level FROM sys.databases;
+```
+
+If the compatibility level is below 130 (SQL Server 2016), consider modifying it to a newer value ([documentation](https://learn.microsoft.com/en-us/sql/t-sql/statements/alter-database-transact-sql-compatibility-level?view=sql-server-ver16)).
```
Comment (ErikEJ) [inline entity-framework/core/what-is-new/ef-core-8.0/breaking-changes.md:70]: https://learn.microsoft.com/sql/t-sql/statements/alter-database-transact-sql-compatibility-level

File: entity-framework/core/what-is-new/ef-core-8.0/breaking-changes.md [hunk 2]
```diff
@@ -14,8 +14,69 @@ This page will document API and behavior changes that have the potential to brea
...
+
+... would be translated to the following SQL:
+
+```sql
+SELECT [b].[Id], [b].[Name]
+FROM [Blogs] AS [b]
+WHERE [b].[Name] IN (N'Blog1', N'Blog2')
+```
+
+Such insertion of constant values into the SQL creates many performance problems, defeating query plan caching and causing unneeded evictions of other queries. The new EF Core 8.0 translation uses the SQL Server [`OPENJSON`](https://learn.microsoft.com/sql/t-sql/functions/openjson-transact-sql) function to instead transfer the values as a JSON array. This solves the performance issues inherent in the previous technique; however, the `OPENJSON` function is unavailable in SQL Server 2014 or below.
```
Comment (ErikEJ) [inline entity-framework/core/what-is-new/ef-core-8.0/breaking-changes.md:58]: "and below" ?

File: entity-framework/core/what-is-new/ef-core-8.0/breaking-changes.md (hunk 1 above)
Comment (ErikEJ) [reply entity-framework/core/what-is-new/ef-core-8.0/breaking-changes.md:70]: Maybe link specifically to this? https://learn.microsoft.com/sql/t-sql/statements/alter-database-transact-sql-compatibility-level#best-practices-for-upgrading-database-compatibility-level

File: entity-framework/core/what-is-new/ef-core-8.0/breaking-changes.md [hunk 3]
```diff
@@ -14,8 +14,69 @@ This page will document API and behavior changes that have the potential to brea
...
+
+#### Mitigations
+
+If your database is SQL Server 2016 (13.x) or newer, or if you're using Azure SQL, check the configured compatibility level of your database via the following command:
+
+```sql
+SELECT name, compatibility_level FROM sys.databases;
+```
+
+If the compatibility level is below 130 (SQL Server 2016), consider modifying it to a newer value ([documentation](https://learn.microsoft.com/en-us/sql/t-sql/statements/alter-database-transact-sql-compatibility-level?view=sql-server-ver16)).
```
Comment (ErikEJ) [inline entity-framework/core/what-is-new/ef-core-8.0/breaking-changes.md:70]: Maybe link specifically to this in a note? https://learn.microsoft.com/sql/t-sql/statements/alter-database-transact-sql-compatibility-level#best-practices-for-upgrading-database-compatibility-level

File: entity-framework/core/what-is-new/ef-core-8.0/breaking-changes.md [hunk 4]
```diff
@@ -33,7 +33,7 @@ Previously, when the `Contains` operator was used in LINQ queries with a paramet
 
 Starting with EF Core 8.0, EF now generates SQL that is more efficient, but is unsupported on SQL Server 2014 and below.
 
-Note that newer SQL Server versions may be configured with an older [compatibility level](https://learn.microsoft.com/en-us/sql/t-sql/statements/alter-database-transact-sql-compatibility-level?view=sql-server-ver16), also making them incompatible with the new SQL. This can also occur with an Azure SQL database which was migrated from a previous on-premises SQL Server instance, carrying over the old compatibility level.
+Note that newer SQL Server versions may be configured with an older [compatibility level](https://learn.microsoft.com/en-us/sql/t-sql/statements/alter-database-transact-sql-compatibility-level), also making them incompatible with the new SQL. This can also occur with an Azure SQL database which was migrated from a previous on-premises SQL Server instance, carrying over the old compatibility level.
```
Comment (ErikEJ) [inline entity-framework/core/what-is-new/ef-core-8.0/breaking-changes.md:36]: Also remove en-us part of url

File: entity-framework/core/what-is-new/ef-core-8.0/breaking-changes.md (hunk 3 above)
Comment (roji) [reply entity-framework/core/what-is-new/ef-core-8.0/breaking-changes.md:70]: Sure, makes sense - thanks.

**Comment by ErikEJ** [comment]
LGTM

**Review by ajcvickers — APPROVED** [review]

**Pull request merged**